tests/unit/sync_strategies/test_binlog.py

```diff
@@ -2037,8 +2084,9 @@ def test_calculate_gtid_bookmark_for_mysql_no_gtid_found_expect_exception(self):
             'bookmarks': {
             }
         }
+        mysql_conn = Mock(spec_set=MySQLConnection)
 
         with self.assertRaises(Exception) as ex:
-            binlog.calculate_gtid_bookmark(binlog_streams, state, connection.MYSQL_ENGINE)
+            binlog.calculate_gtid_bookmark(mysql_conn, binlog_streams, state, connection.MYSQL_ENGINE)
 
             self.assertEqual("Couldn't find any gtid in state bookmarks to resume logical replication", str(ex))
```

Comment:
This part will be passed even if it does not raise an exception!

suggestion:

```python
with self.assertRaises(Exception) as ex:
             binlog.calculate_gtid_bookmark(mysql_conn, binlog_streams, state, connection.MYSQL_ENGINE)

self.assertEqual("Couldn't find any gtid in state bookmarks to resume logical replication", str(ex))
```

Reply:
hmm, if no exception is raised then `self.assertRaises` will fail first

Reply:
Nope!

Reply:
if `binlog.calculate_gtid_bookmark(mysql_conn, binlog_streams, state, connection.MYSQL_ENGINE)` does not raise an exception then the test would fail with `AssertionError: Exception not raised` and would not get to the `self.assertEqual` bit.

Try this:
```
import unittest

class MyTestCase(unittest.TestCase):
    def test_not_raises_exception(self):
        """shoud fail"""
        with self.assertRaises(Exception) as x:
            10/2
            self.assertIsNotNone(x)


if __name__ == '__main__':
    unittest.main()

```

If I'm misunderstanding your point then apologies.

Reply:
Test this one instead, which is same as the main test:

```python
import unittest


class MyTestCase(unittest.TestCase):

    def test_not_raises_exception(self):
        """should fail but passes"""
        with self.assertRaises(Exception) as x:
            10/1
            self.assertEqual('foo', str(x))


if __name__ == '__main__':
    unittest.main()
```

but this one fails as expected:

```python
Test this one instead which is same as the main test:

```python
import unittest


class MyTestCase(unittest.TestCase):

    def test_not_raises_exception(self):
        """shoud fail"""
        with self.assertRaises(Exception) as x:
            10/1
        self.assertEqual('foo', str(x))


if __name__ == '__main__':
    unittest.main()
```

Reply:
Aha, got it.

Reply:
Fixed.